### PR TITLE
Terminate if main window is closed and status indicator is not showing

### LINKED ISSUE #369: Gnome on Wayland: Syncthing-gtk lingers with no windows open and no indicator
Per #343, syncthing-gtk's indicator does not appear when running on Gnome on Wayland. Despite this, when syncthing-gtk's main window is closed, its process does not terminate.

When running under a shell that lacks indicator support, syncthing-gtk should terminate when its window is closed. Also, “minimize to tray on start” (in the UI settings) should be grayed out.

Note that [this Gnome Shell extension](https://extensions.gnome.org/extension/615/appindicator-support/) *adds* indicator support. If that extension is running, syncthing-gtk's indicator will function as normal, so lingering and minimize-to-tray are fine.

### PR DESCRIPTION
**Pull request by argv-minus-one**
As mentioned in #369, when syncthing-gtk cannot show its status indicator and the main window is closed, its process does not terminate.

I looked over your code and found that this situation is not specific to Gnome or Wayland. Syncthing-gtk already can detect whether the status indicator is showing (the `active` property on `StatusIcon`), and it already has a fallback for when the platform doesn't seem to support status indicators at all (`StatusIconDummy`).

I see two problems here:

1. `StatusIconAppIndicator` doesn't update its `active` property.

2. Closing the main window doesn't terminate syncthing-gtk, even if the status indicator is not active.

This PR contains two commits, one for each of the aforementioned problems (f16fa9f and 5d97439, respectively). This directly fixes #369.

### CODE REVIEW

**Comment by kozec** [comment]
Wait, wait, why should closing window mean quit if there is no tray icon visible?

I mean, I could imagine such thing as _option_ , but considering that doing so will terminate syncthing as well, I think this is rather big change in behavior.

**Comment by argv-minus-one** [comment]
Syncthing-gtk only terminates the daemon on exit if that's what the user says to do in the UI settings. The default setting is “ask me what to do”, which should happen as normal if this PR is merged.

In order for this behavior to be surprising, the user would have to have explicitly chosen “terminate daemon”, *and* been reliant on syncthing-gtk continuing to run in the background with no status icon.

On the other hand, it is surprising to me that syncthing-gtk *doesn't* terminate when its main window is closed without leaving behind some artifact (i.e. the status icon). That's what got me started on this all-day adventure through your code base. 😄

I can see one use case for leaving syncthing-gtk in the background even without a status icon: the user wants the syncthing daemon to run while the user is logged in, and to terminate when the user logs out. However, syncthing-gtk doesn't actually support this: when syncthing-gtk terminates because of losing its connection to the window system, the “terminate daemon” setting is ignored.

Hence, I can't think of any reason to leave syncthing-gtk running with no window and no status icon.

**Comment by kozec** [comment]
> However, syncthing-gtk doesn't actually support this: when syncthing-gtk terminates because of losing its connection to the window system, the “terminate daemon” setting is ignored.

That actually sounds like a bug. I mean, when you are logging out, WM is supposed to set `wm_delete_window` on everything and wait for programs to terminate correctly, but I can't blame anyone for logging out with atl+ctrl+backspace.

I'll check if there is way to catch such case.

> In order for this behavior to be surprising, the user would have to have explicitly chosen “terminate daemon”, and been reliant on syncthing-gtk continuing to run in the background with no status icon.

I kinda imagine that's what everyone is doing, checking that "don't ask me again" checkbox and choosing either one option. I think best way to do this is to merge this and add option to UI settings.

Basically, I don't like when program changes behavior between updates :)

**Comment by argv-minus-one** [comment]
Fair enough. I'll think it over when I have more time.
